lib/Provider/future_provider.dart

```diff
@@ -163,18 +164,34 @@ class BrightnessNotifier extends AutoDisposeNotifier<Brightness>
     with WidgetsBindingObserver {
   @override
   void didChangePlatformBrightness() {
-    state = WidgetsBinding.instance.platformDispatcher.platformBrightness;
+    changeBrightness(
+        WidgetsBinding.instance.platformDispatcher.platformBrightness ==
+            Brightness.dark);
     super.didChangePlatformBrightness();
   }
 
   @override
   Brightness build() {
+    bool? isDark = Config().pref.getBool("isDark");
     WidgetsBinding.instance.addObserver(this);
-    return WidgetsBinding.instance.platformDispatcher.platformBrightness;
+    if (isDark == null) {
+      Brightness brightness =
+          WidgetsBinding.instance.platformDispatcher.platformBrightness;
+      brightness == Brightness.dark
+          ? Config().pref.setBool("isDark", true)
+          : Config().pref.setBool("isDark", false);
+      return brightness;
+    } else {
+      return isDark ? Brightness.dark : Brightness.light;
+    }
   }
 
-  void changeBrightness(bool value) =>
-      state = (value) ? Brightness.dark : Brightness.light;
+  Future<void> changeBrightness(bool value) async {
```

Comment:
instead of having one function with bool value being passed in, i think you should make 2 different void functions called setDark and setLight. This will make it easier to use this provider functionality